hazelcast-client/src/main/java/com/hazelcast/client/impl/clientside/FailoverClientConfigSupport.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.clientside;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.config.ClientFailoverConfig;
+import com.hazelcast.client.config.ClientNetworkConfig;
+import com.hazelcast.client.config.XmlClientConfigBuilder;
+import com.hazelcast.client.config.XmlClientConfigLocator;
+import com.hazelcast.client.config.XmlClientFailoverConfigBuilder;
+import com.hazelcast.client.config.YamlClientConfigBuilder;
+import com.hazelcast.client.config.YamlClientConfigLocator;
+import com.hazelcast.config.InvalidConfigurationException;
+
+import java.util.List;
+
+
+/**
+ * Static methods to resolve and validate multiple client configs for blue green feature
+ */
+public final class FailoverClientConfigSupport {
+
+    private FailoverClientConfigSupport() {
+    }
+
+    /**
+     * If clientFailoverConfig passed null, then we try to load config via system property.
+     * If still could no create HazelcastException is thrown.
+     *
+     * @param clientFailoverConfig provided via HazelcastClient.newHazelcastClient(ClientFailoverConfig config)
+     * @return resolvedConfigs
+     * @throws InvalidConfigurationException when given config is not valid
+     */
+    public static ClientFailoverConfig resolveClientConfig(ClientFailoverConfig clientFailoverConfig) {
+        if (clientFailoverConfig == null) {
+            XmlClientFailoverConfigBuilder configBuilder = new XmlClientFailoverConfigBuilder();
+            clientFailoverConfig = configBuilder.build();
+        }
+        checkValidAlternative(clientFailoverConfig.getClientConfigs());
+        return clientFailoverConfig;
+    }
+
+    /**
+     * Returns a ClientFailoverConfig with single client config. If clientConfig is null,
+     * We create it via XmlClientConfigBuilder().build()
+     *
+     * @param config provided via HazelcastClient.newHazelcastClient(ClientConfig config)
+     * @return resolvedConfigs
+     * @throws InvalidConfigurationException when given config is not valid
+     */
+    public static ClientFailoverConfig resolveClientConfig(ClientConfig config) {
+        if (config == null) {
+            config = new XmlClientConfigBuilder().build();
```

Comment:
This should follow the same logic as the below method with looking up for both XML and YAML files. I left it this way with my client-side YAML change, just noticed now.

I think it's not a show-stopper for FF and I can take care of it, just commenting.

Reply:
Looks like this comment is relevant to #14342, not to this PR

Reply:
cc @sancar